integration_tests/src/main/python/join_test.py

```diff
@@ -23,7 +23,9 @@
 all_gen = [StringGen(), ByteGen(), ShortGen(), IntegerGen(), LongGen(),
```

Comment:
Its better to use the `all_gen` in `data_gen.py` and add the stuff missing

Reply:
So I really hate `all_gen` being in data_gen.py because it is not really generating everything (all). I let it through before so I didn't say anything when it was reused, but if you do use it I would want it to be renamed so it is clear what is in it.

Reply:
`all_gen` in `data_gen.py` is missing these 2 gens: `null_gen, decimal_gen_neg_scale` . 
Please let me know if below would be acceptable name if we want to rename it for better readability? 

1. `all_gen_minus_null_gen_and_decimal_gen_neg_scale` OR  `all_gen_excluding_null_gen_and_decimal_gen_neg_scale` (problem I see with this is we may have to rename it again if we add any other gen in the future which is not included in this variable).
Suggestions are welcome on variable name :) .

Or would you prefer to keep `all_gen` local to the test files and remove it from `data_gen.py` ?(I can create follow on PR for that as it involves changing other files).

Reply:
I can't think of a name suitable. @jlowe is better at naming things may be he has something

The right thing to do in my mind would be to add `everything` in all and let tests create local versions of a list if they need something different. Considering it might break tests if you add everything in `all_gen` lets create a follow-on for this as a lower priority task